demos/digest/EVP_MD_demo.c

```diff
@@ -0,0 +1,189 @@
+/*-
+ * Copyright 2019-2020 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+/*
+ * Example of using EVP_MD_fetch and EVP_Digest* methods to calculate
+ * a digest of static buffers
+ */
+
+#include <string.h>
+#include <stdio.h>
+#include <openssl/err.h>
+#include <openssl/evp.h>
+
+/*-
+ * This demonstration will show how to digest data using
+ * the soliloqy from Hamlet scene 1 act 3
+ * The soliloqy is split into two parts to demonstrate using EVP_DigestUpdate
+ * more than once.
+ */
+
+const char * hamlet_1 =
+	"To be, or not to be, that is the question,\n"
+	"Whether tis nobler in the minde to suffer\n"
+	"The ſlings and arrowes of outragious fortune,\n"
+	"Or to take Armes again in a sea of troubles,\n"
+	"And by opposing, end them, to die to sleep;\n"
+	"No more, and by a sleep, to say we end\n"
+	"The heart-ache, and the thousand natural shocks\n"
+	"That flesh is heir to? tis a consumation\n"
+	"Devoutly to be wished. To die to sleep,\n"
+	"To sleepe, perchance to dreame, Aye, there's the rub,\n"
+	"For in that sleep of death what dreams may come\n"
+	"When we haue shuffled off this mortal coil\n"
+	"Must give us pause. There's the respect\n"
+	"That makes calamity of so long life:\n"
+	"For who would bear the Ships and Scorns of time,\n"
+	"The oppressor's wrong, the proud man's Contumely,\n"
+	"The pangs of dispised love, the Law's delay,\n"
+;
+const char * hamlet_2 =
+	"The insolence of Office, and the spurns\n"
+	"That patient merit of the'unworthy takes,\n"
+	"When he himself might his Quietas make\n"
+	"With a bare bodkin? Who would fardels bear,\n"
+	"To grunt and sweat under a weary life,\n"
+	"But that the dread of something after death,\n"
+	"The undiscovered country, from whose bourn\n"
+	"No traveller returns, puzzles the will,\n"
+	"And makes us rather bear those ills we have,\n"
+	"Then fly to others we know not of?\n"
+	"Thus conscience does make cowards of us all,\n"
+	"And thus the native hue of Resolution\n"
+	"Is sickled o'er with the pale cast of Thought,\n"
+	"And enterprises of great pith and moment,\n"
+	"With this regard their currents turn awry,\n"
+	"And lose the name of Action. Soft you now,\n"
+	"The fair Ophelia? Nymph in thy Orisons\n"
+	"Be all my sins remember'd.\n"
+; 
+
+/*
+ * The known value of the SHA3-512 digest of the above soliloqy
+ */
+const unsigned char known_answer[] = {
```

Comment:
Just thinking out loud - I wonder if we should even do the KAT part?
Initially it is useful to see that you are doing the correct thing in the demo code.
Keeping these demos as concise as possible might be a good thing?

Reply:
How do **we** know that the demo is working?

Reply:
It needs to work when you write it.. you can check manually then that it is correct if it prints it out.
I am thinking of the demos not as tests - but as usage examples. A user doesnt do a KAT test on a digest, a test does.

Reply:
I thought about adding a README file that tells the user how to download SHA3_512ShortMsg.rsp test data from NIST and then how to run the EVP_MD_stdin demo to see if its output matches.

Reply:
or we could just keep it simple :)